### PR TITLE
Add API for sync function command

### PR DESCRIPTION
**Pull request by kfurgol**
**Description**

Changes proposed in this pull request:

- Added API for sync function command

**Related issue(s)**
[#9403](https://github.com/kyma-project/kyma/issues/9403)

### CODE REVIEW

File: function/examples/sync/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Synchronise function workspace
+
+## Overview
+
+This example shows you how to synchronise serverless workspace using function api.
```
[the remark this answers is not in the document]
Comment (kazydek) [reply function/examples/sync/README.md:5]: According to the description in [issue 9403](https://github.com/kyma-project/kyma/issues/9403), but correct me if sth has changed.

File: function/examples/sync/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+### Usage
```
Comment (kazydek) [inline function/examples/sync/README.md:7]:
```suggestion
## Usage
```

File: function/examples/sync/README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+```bash
+Usage:
+	sync <name> [ --kubeconfig=<FILE> ] [ --output=<DIR> ] [options]
+
+Options:
+	-n --namespace=<NAMESPACE>  Choose namespace for your function [default: default].
+	--debug                 Enable verbose output.
+	-h --help               Show this screen.
+	--version               Show version.
```
[the remark this answers is not in the document]
Comment (kazydek) [reply function/examples/sync/README.md:17]: Could you even these comments out? (move them a bit to the right)?
image

**Review by kazydek — CHANGES_REQUESTED** [review]

**Review by kazydek — COMMENTED** [review]
Tried to change from "requesting changes" to "comments" in case I'm not the one who will double-check it, but failed :(  Let me know when you add changes and I'll double-check it.

File: function/pkg/workspace/workspace.go [hunk 4]
```diff
@@ -58,6 +83,83 @@ func fromRuntime(runtime types.Runtime) (workspace, error) {
 	}
 }
 
+func Synchronise(ctx context.Context, config Cfg, outputPath string, build client.Build) error {
+	return synchronise(ctx, config, outputPath, build, defaultWriterProvider)
+}
+
+func synchronise(ctx context.Context, config Cfg, outputPath string, build client.Build, writerProvider WriterProvider) error {
```
Comment (tgorgol) [inline function/pkg/workspace/workspace.go:90]: I think that this function could use some refactoring. The whole synchronization logic is in this single function and I'm not even sure if it belongs in the `workspace` package.

**Review by tgorgol — APPROVED** [review]

**Review by tgorgol — APPROVED** [review]

**Review by kazydek — APPROVED** [review]
Approving as docs were moved to a separate [PR](https://github.com/kyma-incubator/hydroform/pull/134).

**Review by m00g3n — APPROVED** [review]

**Pull request merged**